src/client/dfs/dfs.c

```diff
@@ -1798,23 +1799,11 @@ dfs_lookup(dfs_t *dfs, const char *path, int flags, dfs_obj_t **_obj,
 	struct dfs_entry	entry = {0};
 	size_t			len;
 	int			rc;
+	bool			parent_fully_valid;
 
-	if (dfs == NULL || !dfs->mounted)
-		return EINVAL;
-	if (_obj == NULL)
-		return EINVAL;
-	if (path == NULL || strnlen(path, PATH_MAX) > PATH_MAX - 1)
-		return EINVAL;
-	if (path[0] != '/')
-		return EINVAL;
-
-	/** if we added a prefix, check and skip over it */
-	if (dfs->prefix) {
-		if (strncmp(dfs->prefix, path, dfs->prefix_len) != 0)
-			return EINVAL;
-
-		path += dfs->prefix_len;
-	}
+	/* Only paths from root can be absolute */
+	if (path[0] == '/' && daos_oid_cmp(root->oid, dfs->root.oid) != 0)
+ 		return EINVAL;
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs